Ros1 update docs (#122)
husarion/panther_ros: panther_bringup/README.md

# panther_bringup

> Package containing default config and launch files, necessary to  start all base functionalities of the Husarion Panther robot.

Reviewer: ```suggestion
The package containing default configuration and launch files necessary to start all basic functionalities of the Husarion Panther robot.
```

## Default nodes launched

- `battery_node` - node responsible for monitoring and publishing internal battery state of the Husarion Panther robot. For more information refer to: [panther_battery](../panther_battery/README.md).
- `imu_manager` - nodelet managing Phidget Spatial IMU sensor.
- `lights_driver_node` - node responsible for displaying frames on the Husarion Panther robot LED panels. For more information refer to: [panther_lights](../panther_lights/README.md).
- `lights_controller_node` - node responsible for processing animations and publishing frames to be displayed on the Husarion Panther robot LED panels. For more information refer to: [panther_lights](../panther_lights/README.md).
- `manager_bt_node` - node responsible for managing the Husarion Panther robot. Designs a system handling LED panels, safety features and software shutdown of components. For more information refer to: [panther_manager](../panther_manager/README.md).
- `panther_driver` - node responsible for communication with motor controllers and computing inverse and forward kinematics of a robot. For more information refer to: [panther_driver](../panther_driver/README.md).

> - `power_control`  - node responsible for power management of the Husarion Panther robot. For more information refer to: [panther_power_control](../panther_power_control/README.md).

Reviewer: ```suggestion
- `power_control` - node responsible for power management of the Husarion Panther robot. For more information refer to: [panther_power_control](../panther_power_control/README.md).
```

- `robot_state_publisher` - node publishing description of the Husarion Panther robot. For more information refer to: [panther_description](../panther_description/README.md).

## Launch arguments

- `bt_project_file` [*string*, default=**$(find panther_manager)/config/Panther106BT.btproj**]: path to BehaviorTree project used by `panther_manager`. Default value of this parameter depends on `panther_version` argument. For version 1.2 and above it is: **(find panther_manager)/config/Panther106BT.btproj**.
- `disable_manager` [*bool*, default=**false**]: allows to disable `panther_manager`, used for testing purposes.
- `namespace` [*string*, default=**panther**]: namespace to use with robot.
- `panther_common_config_file` [*string*, default=**$(find panther_bringup)/config/panther_common.yaml**]: path to yaml file with standard parameters used by the `driver_node`.
- `panther_version` [*float*, default=**1.0**]: robot version parsed using `PANTHER_ROBOT_VERSION` enironmental variable.
- `publish_robot_state` [*bool*, default=**true**]: whether to publish default Panther robot description.

> - `robot_description` [*string*]: robot description in URDF format, by default parsed from **$(find panther_description)/urdf/panther.urdf.xacro**, with `wheel_config_path` specified by `wheel_config_file` launch argument and imu position parsed using `PANTHER_IMU_LOCALIZATION_X/Y/Z` and `PANTHER_IMU_ORIENTATION_R/P/Y` evironmental variables.

Reviewer: Complicated, maybe creating a list of sub-arguments would be better.

```suggestion
- `robot_description` [*string*]: robot description in URDF format, parsed by default from **$(find panther_description)/urdf/panther.urdf.xacro**. Urdf itself contains its own arguments that are responsible for wheel selection and imu position. 
  - `wheel_config_path` reads the value of the `wheel_config_file` parameter
  - changing the imu position is done by setting the environment variables `PANTHER_IMU_LOCALIZATION_X/Y/Z` and `PANTHER_IMU_ORIENTATION_R/P/Y`.

- `shutdown_hosts_config_file` [*string*, default=**$(find panther_bringup)/config/shutdown_hosts.yaml**]: path to YAML file with description of hosts to shutdown.

> - `test_animations` [*bool*, default=**false**]: if true then something but i forgot what at this point

Reviewer: I believe you will find out. ;)
If I undersed it right [lights readme](https://github.com/husarion/panther_ros/blob/ros1/panther_lights/README.md/#defining-animations) this param turns on test mode, which allow you to test new animations using services.

Author: completely forgot about this :rofl:
